client_src/js/lib/modules/viewUpdater.js

```diff
@@ -1000,17 +1000,15 @@ async function updateState(printer, clientSettings, view) {
       const tool = document.getElementById(`${printer._id}-spool-${i}`);
       if (printer.selectedFilament[i] !== null) {
         const filamentManager = await checkFilamentManager();
-        if (filamentManager) {
-          tool.innerHTML = `${printer.selectedFilament[i].spools.material}`;
-        } else {
+        if (tool) {
           tool.innerHTML = `${printer.selectedFilament[i].spools.material}`;
         }
       } else {
-        tool.innerHTML = "No Spool";
+        if (tool) {
+          tool.innerHTML = "No Spool";
+        }
       }
     }
-  } else {
```

Comment:
This shouldn't even run so it's pointless. There's no tool boxes filled if it doesn't know the actual tool counts from the printer.